test/fake-intake/fakeintake/server.go

```diff
@@ -0,0 +1,153 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package fakeintake
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+)
+
+type payload struct {
+	timestamp time.Time
+	data      []byte
+}
+
+type FakeIntake struct {
+	mu     sync.RWMutex
+	server http.Server
+
+	payloadStore map[string][]payload
+}
+
+// NewFakeIntake creates a new fake intake and starts it on localhost:5000
+func NewFakeIntake() *FakeIntake {
```

Comment:
You may want to offer a `NewFakeIntake(ctx context.Context)` or a `Run(ctx context.Context)` as it will be easier to use when embedded.

Reply:
Is this a blocker, or can I take care of this in a follow up PR ?

Reply:
Will be address by AETT-83